File: UsingOpenNLP/app/src/main/java/nisrulz/github/sample/usingopennlp/opennlp/SentenceDetector.java
package nisrulz.github.sample.usingopennlp.opennlp;

import android.app.Activity;
import android.util.Log;
import java.io.IOException;
import java.io.InputStream;
import nisrulz.github.sample.usingopennlp.R;
import opennlp.tools.sentdetect.SentenceDetectorME;
import opennlp.tools.sentdetect.SentenceModel;

public class SentenceDetector {
  public String[] findSentences(Activity activity, String data) {
    System.out.println("\n>> Running " + getClass().getSimpleName() + "\n");

    InputStream is = null;
    try {

      is = activity.getResources().openRawResource(R.raw.en_sent);

      if (is != null) {
        SentenceModel model = new SentenceModel(is);
        is.close();
        is = null;
        SentenceDetectorME sdetector = new SentenceDetectorME(model);
        String sentences[] = sdetector.sentDetect(data);
        return sentences;
      }
      else {
        Log.i("TAG", "findSentences: NULL");
      }
    } catch (Exception ex) {
      ex.printStackTrace();
    } finally {
      if (is != null) {
        try {
          is.close();
        } catch (IOException e) {
          e.printStackTrace();
        }
      }
    }
    return new String[] {};
  }
}

File: UsingOpenNLP/app/src/main/java/nisrulz/github/example/usingopennlp/opennlp/SentenceDetector.kt
package nisrulz.github.example.usingopennlp.opennlp

import android.app.Activity
import nisrulz.github.example.usingopennlp.R
import opennlp.tools.sentdetect.SentenceDetectorME
import opennlp.tools.sentdetect.SentenceModel
import java.io.IOException
import java.io.InputStream

class SentenceDetector {
    fun findSentences(activity: Activity, data: String?): Array<String> {
        println(
            """
    
    >> Running ${javaClass.simpleName}
    
    """.trimIndent()
        )
        var inputStream: InputStream? = null
        try {
            inputStream = activity.resources.openRawResource(R.raw.en_sent)
            val model = SentenceModel(inputStream)
            inputStream.close()
            inputStream = null
            val sdetector = SentenceDetectorME(model)
            return sdetector.sentDetect(data)
        } catch (ex: Exception) {
            ex.printStackTrace()
        } finally {
            if (inputStream != null) {
                try {
                    inputStream.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }
        }
        return arrayOf()
    }
}